src/predefined-variables.ts

```diff
@@ -50,16 +64,20 @@ export function init ({gitData, argv, envMatchedVariables}: PredefinedVariablesO
         CI_COMMIT_DESCRIPTION: "More commit text",
         CI_DEFAULT_BRANCH: gitData.branches.default,
         CI_PIPELINE_SOURCE: "push",
-        CI_SERVER_FQDN: `${gitData.remote.host}:${CI_SERVER_PORT}`,
-        CI_SERVER_HOST: `${gitData.remote.host}`,
+        CI_SERVER_FQDN: CI_SERVER_FQDN,
+        CI_SERVER_HOST: CI_SERVER_HOST,
         CI_SERVER_PORT: CI_SERVER_PORT,
         CI_SERVER_SHELL_SSH_PORT: CI_SERVER_SHELL_SSH_PORT,
         CI_SERVER_URL: CI_SERVER_URL,
-        CI_SERVER_PROTOCOL: "https",
+        CI_SERVER_PROTOCOL: CI_SERVER_PROTOCOL,
         CI_API_V4_URL: `${CI_SERVER_URL}/api/v4`,
         CI_PROJECT_URL: `${CI_SERVER_URL}/${gitData.remote.group}/${gitData.remote.project}`,
```

Comment:
```suggestion
        CI_API_V4_URL: `${CI_SERVER_FQDN}/api/v4`,
        CI_PROJECT_URL: `${CI_SERVER_FQDN}/${gitData.remote.group}/${gitData.remote.project}`,
```

Reply:
We really do want `CI_SERVER_URL` here. That has the schema prefix (`https://`), whereas `CI_SERVER_FQDN` does not. Applying the suggested change would be incompatible with GitLab and not be backwards compatible.

Reply:
opps, yes you're right